src/ol/format/kml.js

```diff
@@ -1078,6 +1108,9 @@ ol.format.KML.readStyle_ = function(node, objectStack) {
   var imageStyle = /** @type {ol.style.Image} */
       ('imageStyle' in styleObject ?
           styleObject['imageStyle'] : ol.format.KML.DEFAULT_IMAGE_STYLE_);
+  if (imageStyle == ol.format.KML.DEFAULT_NO_IMAGE_STYLE_) {
```

Comment:
Couldn't you initialize the `ol.format.ol.KML.DEFAULT_NO_IMAGE_STYLE_` with `undefined` instead? Then you wouldn't need this if-block.